src/App.jsx

```diff
@@ -14,33 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const sortByLength = 'length';
+const sortByABC = 'abc';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const getSortGoods = (goods, sortField, isReversed) => {
+  const preparedList = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedList.sort((a, b) => {
+      switch (sortField) {
+        case sortByLength:
+          return a.length - b.length;
+        case sortByABC:
+          return a.localeCompare(b);
+        default:
+          return 0;
+      }
+    });
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    preparedList.reverse();
+  }
+
+  return preparedList;
+};
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const finalGoods = getSortGoods(goodsFromServer, sortField, isReversed);
+
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn('button is-info', {
+            'is-light': sortField !== sortByABC,
+          })}
+          onClick={() => {
+            setSortField(sortByABC);
+          }}
+        >
+          Sort alphabetically
+        </button>
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+        <button
+          type="button"
+          className={cn('button is-success', {
+            'is-light': sortField !== sortByLength,
+          })}
+          onClick={() => {
+            setSortField(sortByLength);
+          }}
+        >
+          Sort by length
+        </button>
+
+        <button
+          type="button"
+          className={cn('button is-warning', {
+            'is-light': !isReversed,
+          })}
+          onClick={() => {
+            setIsReversed(!isReversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortField || isReversed) && (
+          <button
+            type="button"
+            className={cn('button', 'is-danger', 'is-light')}
+            onClick={() => {
+              setSortField('');
+              setIsReversed(false);
+            }}
```

Comment:
Move this logic from jsx to the helper reset function and use it here
```suggestion
            onClick={reset}
```